Documentation/migration-scripts/python-scripts/fetching_and_auto_mapping_groups.py

```
@@ -0,0 +1,328 @@
+    for g in groups:
+        gid = g["id"]
+        gname = g["name"]
+
+        if gid in protected:
+            logger.info(f"Skipping protected group: {gname}")
+            continue
+
+        for m in memberships:
+            if m["group_id"] == gid:
+                try:
+                    requests.delete(
```

Ruff (0.15.12)

[error] Probable use of `requests` call without timeout

(S113)